File: demos/java-compat/Freeze/phonebook/ContactFactory.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

class ContactFactory implements Ice.ValueFactory
{
    @Override
    public Ice.Object
    create(String type)
    {
        assert(type.equals("::Demo::Contact"));
        return new ContactI(this);
    }

    void
    setEvictor(Freeze.Evictor evictor)
    {
        _evictor = evictor;
    }

    Freeze.Evictor
    getEvictor()
    {
        return _evictor;
    }

    private Freeze.Evictor _evictor;
}

File: demos/cpp98/Freeze/phonebook/ContactFactory.cpp
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

#include <ContactFactory.h>
#include <PhoneBookI.h>

using namespace std;

ContactFactory::ContactFactory()
{
}

Ice::ObjectPtr
#ifndef NDEBUG
ContactFactory::create(const string& type)
#else
ContactFactory::create(const string&)
#endif
{
    assert(_evictor);
    assert(type == "::Demo::Contact");
    return new ContactI(this);
}

void
ContactFactory::setEvictor(const Freeze::EvictorPtr& evictor)
{
    _evictor = evictor;
}

Freeze::EvictorPtr
ContactFactory::getEvictor() const
{
    return _evictor;
}
